operator_ui/src/screens/EditFeedsManager/EditFeedsManagerScreen.tsx

```diff
@@ -0,0 +1,123 @@
+import { Redirect, useLocation, useHistory } from 'react-router-dom'
+
+import { notifySuccessMsg, notifyErrorMsg } from 'actionCreators'
+import { EditFeedsManagerView } from './EditFeedsManagerView'
+import { ErrorHandler } from 'src/components/ErrorHandler/ErrorHandler'
+import { FormValues } from 'components/Form/FeedsManagerForm'
+import { parseInputErrors } from 'src/utils/inputErrors'
+import { Loading } from 'components/Feedback/Loading'
+import {
+  useFeedsManagersQuery,
+  FEEDS_MANAGERS_QUERY,
+} from 'src/hooks/useFeedsManagersQuery'
+
+export const UPDATE_FEEDS_MANAGER_MUTATION = gql`
+  mutation UpdateFeedsManager($id: ID!, $input: UpdateFeedsManagerInput!) {
+    updateFeedsManager(id: $id, input: $input) {
+      ... on UpdateFeedsManagerSuccess {
+        feedsManager {
+          id
+          name
+          uri
+          publicKey
+          jobTypes
+          isBootstrapPeer
+          isConnectionActive
+          bootstrapPeerMultiaddr
+          createdAt
+        }
+      }
+      ... on NotFoundError {
+        message
+        code
+      }
+      ... on InputErrors {
+        errors {
+          path
+          message
+          code
+        }
+      }
+    }
+  }
+`
+
+export const EditFeedsManagerScreen: React.FC = () => {
+  const history = useHistory()
+  const location = useLocation()
+  const dispatch = useDispatch()
+  const { data, loading, error } = useFeedsManagersQuery()
+  const [updateFeedsManager] = useMutation<
+    UpdateFeedsManager,
+    UpdateFeedsManagerVariables
+  >(UPDATE_FEEDS_MANAGER_MUTATION, {
+    refetchQueries: [FEEDS_MANAGERS_QUERY],
+  })
+
+  if (loading) {
+    return <Loading />
+  }
+
+  if (error) {
+    return <ErrorHandler error={error} />
+  }
+
+  // We currently only support a single feeds manager, but plan to support more
+  // in the future.
+  const manager =
+    data != undefined && data.feedsManagers.results[0]
+      ? data.feedsManagers.results[0]
+      : undefined
+
+  if (!manager) {
+    return (
+      <Redirect
+        to={{
+          pathname: '/feeds_manager/new',
+          state: { from: location },
+        }}
+      />
+    )
+  }
+
+  const handleSubmit = async (
+    values: FormValues,
+    { setErrors }: FormikHelpers<FormValues>,
+  ) => {
+    try {
+      const result = await updateFeedsManager({
+        variables: { id: manager.id, input: { ...values } },
+      })
+
+      const payload = result.data?.updateFeedsManager
+      switch (payload?.__typename) {
+        case 'UpdateFeedsManagerSuccess':
+          history.push('/feeds_manager')
+
+          dispatch(notifySuccessMsg('Feeds Manager Updated'))
+
+          break
+        case 'NotFoundError':
+          dispatch(notifyErrorMsg(payload.message))
+
+          break
+        case 'InputErrors':
+          dispatch(notifyErrorMsg('Invalid Input'))
+
+          setErrors(parseInputErrors(payload))
+
+          break
+      }
+    } catch (e) {
+      // TODO - Handle errors
```

Comment:
I'm not really sure what to do with this error on a mutation. It is typed as unknown whereas I would have thought it would be typed as an `ApolloError`

Reply:
There are a lot more possible error types that can show up here. The recommended approach is to check for an error type and have some generic error handling. E.g.
```
if (e instanceof ApolloError) {
 // handle apollo error; usually showing something to the user
} else {
  // handle other possible errors
  console.error(e)
  }
```

Reply:
Updated with error handling